scripts/native-system-benchmark.sh

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env bash
+
+RR='rr record --'
+GDB='gdb -ex run --args'
+VALGRIND='valgrind --leak-check=full'
+DBG="${DBG:-$GDB}"
+
+# runs for DURATION seconds
+# if DURATION is set to 0/inf/infinity, run indefinitely
```

Comment:
This was just a stylistic choice because a benchmark duration of `0` seconds is effectively just a test that the system comes up correctly, and is otherwise non-sensical. I'm not bothered by allowing that use-case though if it feels helpful.